backend/api/generate_api.sh

```diff
@@ -21,124 +21,5 @@
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/job.swagger.json \
-  -A job \
-  --principal models.Principal \
-  -c job_client \
-  -m job_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/run.swagger.json \
-  -A run \
-  --principal models.Principal \
-  -c run_client \
-  -m run_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/experiment.swagger.json \
-  -A experiment \
-  --principal models.Principal \
-  -c experiment_client \
-  -m experiment_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/pipeline.upload.swagger.json \
-  -A pipeline_upload \
-  --principal models.Principal \
-  -c pipeline_upload_client \
-  -m pipeline_upload_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/pipeline.swagger.json \
-  -A pipeline \
-  --principal models.Principal \
-  -c pipeline_client \
-  -m pipeline_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/visualization.swagger.json \
-  -A visualization \
-  --principal models.Principal \
-  -c visualization_client \
-  -m visualization_model \
-  -t ${DIR}/go_http_client
-
-${SWAGGER_CMD} generate client \
-  -f ${DIR}/swagger/healthz.swagger.json \
-  -A healthz \
-  --principal models.Principal \
-  -c healthz_client \
-  -m healthz_model \
-  -t ${DIR}/go_http_client
-
-# Hack to fix an issue with go-swagger
-# See https://github.com/go-swagger/go-swagger/issues/1381 for details.
-sed -i -- 's/MaxConcurrency int64 `json:"max_concurrency,omitempty"`/MaxConcurrency int64 `json:"max_concurrency,omitempty,string"`/g' ${DIR}/go_http_client/job_model/api_job.go
-sed -i -- 's/IntervalSecond int64 `json:"interval_second,omitempty"`/IntervalSecond int64 `json:"interval_second,omitempty,string"`/g' ${DIR}/go_http_client/job_model/api_periodic_schedule.go
-sed -i -- 's/MaxConcurrency string `json:"max_concurrency,omitempty"`/MaxConcurrency int64 `json:"max_concurrency,omitempty,string"`/g' ${DIR}/go_http_client/job_model/api_job.go
-sed -i -- 's/IntervalSecond string `json:"interval_second,omitempty"`/IntervalSecond int64 `json:"interval_second,omitempty,string"`/g' ${DIR}/go_http_client/job_model/api_periodic_schedule.go
-
-# Executes the //go:generate directives in the generated code.
-go generate ./...
-
-# Add license to files in go_http_client.
-find ${DIR}/go_http_client/ -name "*.go" -exec ${AUTOGEN_CMD} -i --no-tlc -c "Google LLC" -l apache {} \;
-
-# Finally, run gazelle to add BUILD files for the generated code.
-bazel run //:gazelle
-# HACK: remove unnecessary BUILD.bazels
-rm -f "$REPO_ROOT/sdk/python/kfp/components/structures/BUILD.bazel" "$REPO_ROOT/tools/metadatastore-upgrade/BUILD.bazel"
+cd ../..
+docker run  --interactive --rm --mount type=bind,source="$(pwd)",target=/app/pipelines buildapi app/pipelines/backend/api/generator.sh
```

Comment:
Will generated code get owned by root admin?

Reply:
I don't follow fully, could you elaborate?

Reply:
If I recall correctly, docker run as root by default, so when writing files in mounted folder, they are owned by root, not current user.
Did it happen when you try this?

Reply:
For me it works fine, but according to this [blog post](https://vsupalov.com/docker-shared-permissions/) it could be due to that I am using mac that it works. 
``` NOTE: if you’re using something like docker on mac, you won’t run into those permission issues, as the file sharing is done through NFS and your local files will have the right user. ```
They mention that the proper solution is to build it specifically for each machine to get the user id and group id correct. I will resercah the topic a bit more to verify there statement and see what would suit best.

Reply:
Yes, that's the problem I was talking about.
Feel free to use any stopgap solution that works well enough, probably we can just call `chown` commands after generating those files.

the solution in the blog post also seems problematic, because it requires building a local image per each user.